pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -1034,6 +1346,9 @@ def __init__(
         self._padding_right = 2 ** len(wires) - 2 - self._padding_left
         self._CX = None
 
+        self.hyperparameters["work_wires"] = self._work_wires
+        self.hyperparameters["control_values"] = self.control_values
```

Comment:
I needed similar changes in the matrix PR, so we'll have a clash. 

Food for thought: There I argued that work_wires do not determine the transformation, just the implementation, and should therefore be stored as attributes `self.work_wires` instead of as hyperparameters. But now I see that we'll loose the ability to use it in decomposition, where it does get important...

Reply:
I thought about it some more and I convinced myself that `work_wires` is _not_ a hyperparameter of the operator. If it wasn't given, one could still clearly define how the operator transforms a state vector. Even if it is more work, I'd treat it as an attribute that gets fed to `compute_decomposition` _besides_ params and hyperparams... what do you think?